manual_hook.rb

```diff
@@ -2,56 +2,74 @@
 
 require 'resolv'
 
-def setup_dns(domain, txt_challenge)
-  resolved = false;
-  singleLoop = false;
-  dns = Resolv::DNS.new;
-  acme_domain = "_acme-challenge."+domain; 
-  puts "Checking for pre-existing TXT record for the domain: \'#{acme_domain}\'."
-
-  until resolved
-    dns.each_resource(acme_domain, Resolv::DNS::Resource::IN::TXT) { |resp|
-     if resp.strings[0] == txt_challenge
-       puts "Found #{resp.strings[0]}. match."
-       resolved = true;
-     else
-       puts "Found #{resp.strings[0]}. no match."
-     end
-    }
-
-    if !resolved
-     if !singleLoop
-       puts "Create TXT record for the domain: \'#{acme_domain}\'. TXT record:"
-       puts "\'#{txt_challenge}\'"
-       puts "Press enter when DNS has been updated..."
-       $stdin.readline()
-       singleLoop = true
-     end
-
-     puts "Didn't find a match for #{txt_challenge}"; 
-     puts "Waiting to retry..."; 
-     sleep 30; 
+# Struct to store a challenge
+Challenge = Struct.new(:domain, :acme_domain, :txt_challenge)
+
+
+# Check if a challenge is resolved, returns true in that case
+def resolved?(dns, challenge)
+  dns.each_resource(challenge[:acme_domain], Resolv::DNS::Resource::IN::TXT) { |resp|
+    if resp.strings[0] == challenge[:txt_challenge]
```

Comment:
Still need to loop over resp.strings when an acme_domain has multiple challenges (aka, fqdn and *.fqdn)

Reply:
Confirming my suspicion in https://github.com/jbjonesjr/letsencrypt-manual-hook/issues/10#issuecomment-390383142